src/librustc/middle/liveness.rs

```diff
@@ -1658,10 +1658,12 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
         if let Some(name) = self.should_warn(var) {
             if is_argument {
                 self.ir.tcx.lint_hir(lint::builtin::UNUSED_ASSIGNMENTS, hir_id, sp,
-                    &format!("value passed to `{}` is never read", name));
+                    &format!("value passed to `{}` is never read
+                     (maybe it is overwritten before being read)", name));
             } else {
                 self.ir.tcx.lint_hir(lint::builtin::UNUSED_ASSIGNMENTS, hir_id, sp,
-                    &format!("value assigned to `{}` is never read", name));
+                    &format!("value assigned to `{}` is never read
+                     (maybe it is overwritten before being read)", name));
```

Comment:
Hmm, it's a nice idea to make a suggestion like this. It'd be nicer if we could find the overwriting assignment. But i'm reluctant to suggest trying to implement that, since this liveness code is ripe to be rewritten completely.

Still, I think that including the suggestion in a parenthetical makes it less likely to be noticed, and more likely to wrap around on the terminal. 

A middle option would be to create a "help". You could change this code from `self.ir.tcx.lint_hir(..)` to:

```rust
let msg = self.ir.tcx.self.struct_span_lint_hir(...)
  .help("maybe it is overwritten before being read?")
  .emit();
```

and I think things would look better.

Reply:
Sounds good to me.

I could try rewriting the liveness code, but I don't think I know enough about how it works...

Reply:
I'd love to talk you through it =) but it'd obviously be a rather bigger endeavor :)

Reply:
I'm happy to try :)